src/hooks/useTracking.js

```diff
@@ -114,11 +114,12 @@ export const useTracking = (
       /* eslint-enable camelcase */
     }
 
-    sendEvent(SCHEMA_NAME, SCHEMA_REV, language, event)
+    if (getConsentStatus()) {
+      sendEvent(SCHEMA_NAME, SCHEMA_REV, language, event)
+    }
   }
 
   useEffect(() => {
```

Comment:
We should return early (at the very beginning os `useTracking` if the consent is `no`. There is no point running all the code then.